File: Sort/462. Minimum Moves to Equal Array Elements II.cpp
class Solution {
public:
    int minMoves2(vector<int>& nums) {
        nth_element(nums.begin(), nums.begin()+nums.size()/2, nums.end());
        int std = 0, median = *(nums.begin()+nums.size()/2);
        for(auto i: nums)
            std += max(i - median, median - i);
        return std;
    }
};

File: Sort/462. Minimum Moves to Equal Array Elements II.py
def minMoves2(self, nums):
    median = sorted(nums)[len(nums) / 2]
    return sum(abs(num - median) for num in nums)

def minMoves2(self, nums):
    nums.sort()
    return sum(nums[~i] - nums[i] for i in range(len(nums) / 2))
